otelconfig/otelconfig.go

```diff
@@ -625,8 +625,7 @@ func ConfigureOpenTelemetry(opts ...Option) (func(), error) {
 	for _, setup := range []setupFunc{setupTracing, setupMetrics} {
 		shutdown, err := setup(c)
 		if err != nil {
-			c.Logger.Fatalf("setup error: %v", err)
-			continue
+			return otelConfig.Shutdown, fmt.Errorf("setup error: %w", err)
```

Comment:
I agree returning an error is better than crashing. Do you think it would be worth capturing all the errors from the for loop, combing them and returning a single error?

Reply:
I think returning early is fine. Thanks for putting together 👍🏻